lib/rspec/rails/extensions/active_support/json.rb

```diff
@@ -0,0 +1,63 @@
+if defined?(::RSpec::Mocks)
+  # Need to require the base file to setup internal support require helpers
+  require 'rspec/mocks'
+  require 'rspec/mocks/test_double'
+  ::RSpec::Mocks::TestDouble.module_exec do
+    # @private
+    def as_json(*args, &block)
+      return method_missing(:as_json, *args, &block) unless null_object?
```

Comment:
It won't. Rails adds `as_json` and `to_json` onto `Object`. That will get called before `method_missing`.